plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -147,56 +187,142 @@ func (p *pusher) start() {
 		}
 	}()
 
-	for {
-		select {
-		case e := <-ec:
-			// Start timer when first event of the batch is added (happens after a flush timer timeout)
-			if len(p.events) == 0 {
-				p.resetFlushTimer()
-			}
+	if p.flagPersistentQueue {
+		for {
```

Comment:
IMO these new `for` blocks should be pulled out into separate functions to simplify the `start` function code